api/app/controllers/api/v1/collections/relationships/resources_controller.rb

```diff
@@ -8,8 +8,9 @@ class ResourcesController < ApplicationController
           before_action :set_collection, only: [:index]
 
           resourceful! Resource, authorize_options: { except: [:index, :show] } do
-            @collection.resources
-                       .query(with_pagination!(resource_filter_params))
+            filter_params =
+              with_pagination!(resource_filter_params).merge(collection: @collection.id)
+            Resource.filter(with_pagination!(filter_params))
```

Comment:
Look at the changes I made to how filtering works. With those changes, I think it should be possible to change this call to look like this:

`Resource.filter(with_pagination!(filter_params), @collection.resources)`

I think this is a cleaner solution, as we can use the association scope rather than fudging the params. Try it and confirm that it works.